api/src/referenceApp.Azure/ActiveDirectoryService.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Security;
+using referenceApp.Azure.Models;
+using Microsoft.Extensions.Options;
+using Microsoft.Identity.Client;
+using referenceApp.Azure.Constants;
+
+namespace referenceApp.Azure
+{
+    public class ActiveDirectoryService : IActiveDirectoryService
+    {
+        private readonly IOptions<ActiveDirectoryConfigModel>? _adOptions;
+
+        public ActiveDirectoryService(IOptions<ActiveDirectoryConfigModel> adOptions)
+        {
+            if (ConfigurationValidatorUtility.ValidateActiveDirectoryConfig(adOptions) == null)
+                _adOptions = adOptions ?? throw new ArgumentNullException(nameof(adOptions));
+        }
+
+        #region Properties
+
+
+        public string ClientId { get { return _adOptions?.Value?.ClientId ?? string.Empty; } }
+
+
+        #endregion
+
+        #region Implementation Methods
+
+        public string? GetAccessToken()
```

Comment:
Why not use async / await instead of calling methods with ExecuteAsync().Result?